evaluation/agent_bench/run_infer.py

```diff
@@ -0,0 +1,436 @@
+        if get_ground_truth_cmd is not None:
+            scpt_name = os.path.join('.', f'{instance.instance_id}_get_ground_truth.sh')
+            host_scpt_name = os.path.join(instance_workspace, scpt_name)
+            create_sh_file(host_scpt_name, get_ground_truth_cmd)
+            logger.info(f'Running get ground truth cmd: {scpt_name}')
+            _, final_ans = sandbox.execute(scpt_name)
+
+    comparison_method = instance.comparison_method
+    logger.info(
+        f'Final message: {agent_answer} | Ground truth: {final_ans} | Comparison method: {comparison_method}'
+    )
+    test_result = compare_results(comparison_method, agent_answer, final_ans)
+
+    if hasattr(config, 'is_mock') and config.is_mock:
+        histories = ['mock history']
+    else:
+        histories = [
+            (event_to_dict(action), event_to_dict(obs)) for action, obs in state.history
+        ]
+
+    # Save the output
+    output = {
+        'instance_id': inst_id,
+        'instance': instance.to_dict(),
+        'instruction': instruction,
+        'metadata': metadata,
+        'history': histories,
+        'error': state.error if state and state.error else None,
+        'test_result': {
+            'agent_answer': agent_answer,
+            'final_answer': final_ans,
+            'check_method': comparison_method,
+            'result': test_result,
+        },
+    }
+
+    # clean up
+    if os.path.exists(instance_workspace):
+        shutil.rmtree(instance_workspace)
+    # Close the sandbox
+    try:
+        sandbox.close()
+    except docker.errors.NotFound as e:
+        logger.error(f'Failed to close sandbox: {e}')
+    return output
+
+
+def find_subdirectories(directory):
+    return [
+        name
+        for name in os.listdir(directory)
+        if os.path.isdir(os.path.join(directory, name))
+    ]
+
+
+def load_datasets_from_local(key: str, _dir: str) -> [dict]:
+    _datasets = {}
+    if os.path.exists(_dir):
+        for filename in os.listdir(_dir):
+            if filename.endswith('.json'):  # assuming the datasets are in csv format
+                file_path = os.path.join(_dir, filename)
+                with open(file_path, 'r') as r:
+                    _datasets[filename] = json.load(r)
+    return flatten(key, _datasets)
+
+
+def flatten(key: str, _data: dict) -> [dict]:
+    flat_data = []
+    for subkey, values in _data.items():
+        if isinstance(values, list):
+            for i in range(len(values)):
+                value = values[i]
+                value['instance_id'] = f'{key}_{subkey}_{i}'
+                value['task_idx'] = key
+                flat_data.append(value)
+    return flat_data
+
+
+if __name__ == '__main__':
+    # =============================================
+    # load datasets
+    # =============================================
+
+    dataset = load_dataset('iFurySt/AgentBench')
+    agent_bench_tests = dataset['osbench'].to_pandas()
```

Comment:
Should we add `osbench` as an argument to `run_infer.sh`? Maybe this can help when we support other tasks in `AgentBench`.

Reply:
Seems @iFurySt 's plan is to only support `osbench` for now, so we could defer this I guess.